benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -480,14 +495,19 @@ public void terminateTest(String testID) {
             Future abortRunningTestTaskFuture = taskExecutorService.submit(abortRunningTestTask);
 
             try {
+
               abortRunningTestTaskFuture.get();
+
             } catch (InterruptedException | ExecutionException e) {
               // nothing to do
               e.printStackTrace();
             }
 
           }
 
+          // set to TERMINATING and wait for experiment to terminate
```

Comment:
Probably this should happen before calling the `AbortRunningTask` because otherwise you could get the result from the `experiment-manager` before the switch to the `TERMINATING` state.